packages/synapse-react-client/src/components/EntityDownloadButton/EntityDownloadButton.tsx

```diff
@@ -0,0 +1,121 @@
+import { DownloadOutlined as DownloadIcon } from '@mui/icons-material'
+import { EntityType } from '@sage-bionetworks/synapse-client'
+
+enum DownloadAction {
+  downloadFile,
+  addToCart,
+  programmaticAccess,
+  programmaticAccessDocker,
+  exportTable,
+}
+
+// Function that creates menu items for download actions
+function getMenuItemForAction(
+  entityId: string,
+  entityName: string,
+  entityType: EntityType,
+  downloadAction: DownloadAction,
+): DropdownMenuItem {
+  switch (downloadAction) {
+    case DownloadAction.downloadFile:
+      return {
+        text: 'Download File',
+        onClick: () => {
+          console.log('Download file:', entityId, entityName)
+          // TODO: Implement file download functionality
+        },
+        tooltipText: 'Download this file directly',
+      }
+    case DownloadAction.addToCart:
+      return {
+        text: 'Add to Download Cart',
+        onClick: () => {
+          console.log('Add file(s) to cart:', entityId)
+          // TODO: Implement add to cart functionality
+        },
+        tooltipText: 'Add file(s) to your download cart',
+      }
+    case DownloadAction.programmaticAccess:
+      return {
+        text: 'Programmatic Access',
+        onClick: () => {
+          console.log('Show programmatic access for:', entityId)
+          // TODO: Implement programmatic access options
+        },
+        tooltipText: 'View programmatic access options',
+      }
+    case DownloadAction.programmaticAccessDocker:
+      return {
+        text: 'Programmatic Access (Docker)',
+        onClick: () => {
+          console.log('Show programmatic access (Docker) for:', entityId)
+          // TODO: Implement programmatic access (Docker) options
+        },
+        tooltipText: 'View programmatic options to pull Docker image',
+      }
+    case DownloadAction.exportTable:
+      return {
+        text: 'Export Table',
+        onClick: () => {
+          console.log('Export table for:', entityId)
+          // TODO: Implement export table functionality
+        },
+        tooltipText: 'Export table data',
+      }
+  }
+}
+
+// Function that returns DropdownMenuItem
+function getDownloadActionsForEntityType(type: EntityType): DownloadAction[][] {
+  switch (type) {
+    case 'file':
+      return [
+        [DownloadAction.downloadFile],
+        [DownloadAction.addToCart, DownloadAction.programmaticAccess],
+      ]
+    case 'project':
+    case 'folder':
+      return [[DownloadAction.addToCart, DownloadAction.programmaticAccess]]
+    case 'entityview':
+    case 'dataset':
+    case 'table':
+      return [[DownloadAction.addToCart, DownloadAction.programmaticAccess]]
+    case 'dockerrepo':
+      return [[DownloadAction.programmaticAccessDocker]]
+    default:
+      return [[DownloadAction.exportTable, DownloadAction.programmaticAccess]]
+  }
+}
+
+export function EntityDownloadButton(props: {
+  entityId: string
+  name: string
+  entityType: string
+}) {
```

Comment:
entityType should be strongly typed to the enum when possible

```suggestion
export function EntityDownloadButton(props: {
  entityId: string
  name: string
  entityType: EntityType
}) {
```